Add data preparation module for Databricks notebook

awaismirza92 opened the pull request:
Closes https://github.com/getml/getml-demo/issues/62

awaismirza92 committed:
Add changes from data ingestion work.

This is a squashed commit of the following:

commit 00f30c5f301f07dce8d399ef872aeab9be0f3ced
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Mon Dec 15 17:55:59 2025 +0100

    Use exact version specifications for databricks dependencies in uv.lock

commit 447d5276c9d50e8b16809aa43e16f1c4cc568d71
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Mon Dec 15 17:55:45 2025 +0100

    Remove comments leaking LLM focus from ingestion module

commit da014151d8bbad3b96ab241ffc9b30f5bb093786
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Mon Dec 15 17:18:41 2025 +0100

    Use exact matches for databricks group in pyproject.toml

commit e46fe23bbc02fea54b2959aec240cc530a7a4fcf
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Mon Dec 15 17:15:44 2025 +0100

    Close spark session if the function creates it

commit 962be5b308fd7519b6ca17faccca87c9c2ecc8e6
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Mon Dec 15 16:56:26 2025 +0100

    Remove trailing blank lines at end of pyproject.toml

commit e009302431a1865000cbbd5317a18792dbb142b5
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Mon Dec 15 16:53:35 2025 +0100

    Remove ignore comment for model_config in TableConfig

commit c4a09559159626a123b84e14cbf715aa57a5c882
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Mon Dec 15 16:49:36 2025 +0100

    Move logging suppression to a dedicated function

commit c006495ee94cca93e71ce520c925c695007d4b07
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Mon Dec 15 16:37:01 2025 +0100

    Add debug logging for successful volume file cleanup

commit 0a19854fe02d8dac47079b427520e5f6c6620fbe
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Mon Dec 15 16:15:35 2025 +0100

    Use module level constant for request timeout & increase it to 300s

commit 375adfd263b8f1746a79288ff0021d7ab46a1fce
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Mon Dec 15 16:06:06 2025 +0100

    Remove the mutation of `os.environ`

commit b3af7fccbb7c5a9cd26de7b5067deb59e0f6c038
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Fri Dec 12 12:37:34 2025 +0100

    Suppress INFO/WARNING from absl/glog

commit f86c97609779c748efc47c7b85c2be6264bb4c47
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Thu Dec 11 18:28:22 2025 +0100

    Fix bug in string wrapping

commit 1fcbd4fc55afefd143534a806deeaff4aef426b5
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Wed Dec 10 17:16:42 2025 +0100

    Rename variable name for clarity

commit 0a780e33337f22f004ab6785818059a236166fe4
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Wed Dec 10 17:11:10 2025 +0100

    Add parameter markers and pydantic validation to avoid SQL injection

commit 1011f94cf66bc02322cc5e9feac495400b346796
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Wed Dec 10 12:00:50 2025 +0100

    Document DEFAULT profile, remove python & preparation

commit dc8691635c47aad9008044ed09a2bf09bd919364
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Tue Dec 9 22:47:08 2025 +0100

    Remove pandas by streaming files to volume

commit 1eb9a88ade2c1d8c4bdd5dd39335edf6aa8964d5
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Tue Dec 9 11:39:08 2025 +0100

    Replace relative import with absolute import in Databricks ingestion

commit 646455950526d26e0f25e727b4bbb5e40a62f553
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Wed Dec 3 14:10:58 2025 +0100

    Migrate to pyproject.toml & uv

commit 7c91e7b1f3de6b80ce136b984118d675cf5a7f41
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Tue Dec 2 16:39:22 2025 +0100

    Mention import of injection function

commit f32956618a7899c4d305093c974769f7efdfb5f8
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Tue Dec 2 16:06:44 2025 +0100

    Make the injection function callable

commit aa9a41014cc5647a5e82789d3dea28273687ab8f
Author: Awais Mirza <47096947+awaismirza92@users.noreply.github.com>
Date:   Tue Dec 2 15:05:55 2025 +0100

    Add Databricks ingestion script and requirements file

awaismirza92 committed:
Add data preparation module for databricks notebook

awaismirza92 committed:
Merge branch '50-data-ingestion-for-databricks-notebook' into 62-add-data-preparation-module-for-databricks-notebook

srnnkls reviewed (changes requested):
I'm not giving the same review two times. If you have common functionality to extract, please go ahead and do so. It is not my role as a reviewer to teach you the most basic software engineering best practices (DRY). The idea of separating database-specific logic from orchestration was giving you the ability to look for a common interface, not pseudo-applying lookalike enterprise patterns to simple data preparation scripts for notebook showcases.

See my review on #58 for detailed feedback. The same issues apply here.

Per #42, the target structure is:

```
getml-demo/integration/{platform}/
├── data/
│   ├── ingestion.py
│   ├── preparation.py
│   └── sql/
│       ├── create_weekly_stores.sql # should probably be called create_stores_per_week or something similar
│       └── calculate_target.sql # should probably called create_weekly_total_sales_per_store_per_week or something similar
└── tests/
    └── test_data_pipeline.py
```

Specific issues:
- Remove _sql_loader.py (just inline the two lines)
- Remove models.py (define the data structures inline)
- Remove custom exceptions (DataPreparationError adds nothing)
- Consolidate 9 SQL files into 2
- Remove display/analysis functions from preparation.py (this is a data loader, not a reporting tool)
- Remove docstrings and ASCII banner logging

awaismirza92 committed:
Remove SQL loading module and display functions & SQL files

awaismirza92 committed:
Remove remaining sql files to reduce total to 2

awaismirza92 committed:
Remove DataPreparationError class and related exception handling

awaismirza92 commented on integration/databricks/data/models.py:
@srnnkls I am confused about "Remove models.py (define the data structures inline)" part. The models defined in`models.py` are being used both by `ingestion.py` and `preparation.py`. If I remove`models.py`, I will have to define them in both modules. Is it okay?

awaismirza92 replied:
`models.py` will be kept as discussed in the daily today.

awaismirza92 committed:
Remove banners & reduce docstrings

awaismirza92 committed:
Rename sql scripts & follow target structure of directories

awaismirza92 committed:
Remove remaining banner logging

awaismirza92 committed:
Merge branch '50-data-ingestion-for-databricks-notebook' into 62-add-data-preparation-module-for-databricks-notebook

awaismirza92 committed:
Move left over changes from PR #51 (commit: bcbc393)

awaismirza92 committed:
Remove a dangling ASCII banner

awaismirza92 commented:
@srnnkls the concerns have been addressed. Have a look again.

---

> Per #42, the target structure is:
> 
> ```
> getml-demo/integration/{platform}/
> ├── data/
> │   ├── ingestion.py
> │   ├── preparation.py
> │   └── sql/
> │       ├── create_weekly_stores.sql # should probably be called create_stores_per_week or something similar
> │       └── calculate_target.sql # should probably called create_weekly_total_sales_per_store_per_week or something similar
> └── tests/
>     └── test_data_pipeline.py
> ```

Fixed: https://github.com/getml/getml-demo/pull/63/changes/f7f21d3fae2a38ed8904a9ff4bb9a5494802e5bf

---

> 
> Specific issues:
> 
> * Remove _sql_loader.py (just inline the two lines)

Removed: https://github.com/getml/getml-demo/pull/63/changes/e70b99a5930decd9b0c2683e3628eff178515702

---

> * Remove models.py (define the data structures inline)

Discussed in today's daily to keep it.

---

> * Remove custom exceptions (DataPreparationError adds nothing)

Removed: https://github.com/getml/getml-demo/pull/63/changes/4a7b1d06b4f52a6f28329b2f2f6099cc29a51358

---

> * Consolidate 9 SQL files into 2

Consolidated: https://github.com/getml/getml-demo/pull/63/changes/e70b99a5930decd9b0c2683e3628eff178515702 & https://github.com/getml/getml-demo/pull/63/changes/25925c99b1a77a0b0143a5d8523edfb3ea549eb0

---

> * Remove display/analysis functions from preparation.py (this is a data loader, not a reporting tool)

Removed: https://github.com/getml/getml-demo/pull/63/changes/e70b99a5930decd9b0c2683e3628eff178515702

---


> * Remove docstrings and ASCII banner logging

Removed: https://github.com/getml/getml-demo/pull/63/changes/2637a337faf4cfd588bda1c1e0f5cddfdecdf5e4 & https://github.com/getml/getml-demo/pull/63/changes/27a3ccdcbadab8c8e0d5e9cc77d81f2ca8b431e6 & https://github.com/getml/getml-demo/pull/63/changes/fd10d9f2ef1a1ec3b191ac4af1d9509c7eef36c6
